frontend/packages/kubevirt-plugin/src/components/modals/clone-vm-modal/clone-vm-modal.tsx

```diff
@@ -233,24 +234,42 @@ export type CloneVMModalProps = CloneVMModalFirehoseProps &
 
 const CloneVMModalFirehose: React.FC<CloneVMModalFirehoseProps> = (props) => {
   const { vm, useProjects } = props;
-  const [namespace, setNamespace] = React.useState(getNamespace(vm));
+  const vmNamespace = getNamespace(vm);
+  const [namespace, setNamespace] = React.useState(vmNamespace);
 
   const requestsDataVolumes = !!getVolumes(vm).find(getVolumeDataVolumeName);
   const requestsPVCs = !!getVolumes(vm).find(getVolumePersistentVolumeClaimName);
 
-  const resources = [
-    getResource(useProjects ? ProjectModel : NamespaceModel, { prop: 'namespaces' }),
-    getResource(VirtualMachineModel, { namespace, prop: 'virtualMachines' }),
+  const resources: FirehoseResource[] = [
+    {
+      kind: (useProjects ? ProjectModel : NamespaceModel).kind,
+      isList: true,
+      prop: 'namespaces',
+    },
+    {
+      kind: VirtualMachineModel.kind,
+      namespace,
```

Comment:
you want to get VMs from target namespace where user will be cloning the VM so you need to update the polling whenever the target namespace changes

Reply:
the namespace is updated accordingly

please see (few lines above)

```tsx
const [namespace, setNamespace] = React.useState(vmNamespace);
```

Reply:
sorry about that :) You're right